org.eclipse.lsp4e/src/org/eclipse/lsp4e/ResourceForUriCache.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   See git history
+ *******************************************************************************/
+
+package org.eclipse.lsp4e;
```

Comment:
This should move it into the internal package. I don't think this should be part of the LSP4E API.